session/src/main/java/org/apache/iotdb/session/Session.java

```diff
@@ -201,6 +206,98 @@ public TSExecuteBatchStatementResp insertBatch(RowBatch rowBatch)
     }
   }
 
+  /**
+   * use batch interface to insert data
+   *
+   * @param rowBatch data batch
+   */
+  public TSExecuteBatchStatementResp insertBatch(RowBatch rowBatch)
+      throws IoTDBSessionException {
+    sortRowBatch(rowBatch);
+
+    return insertSortedBatch(rowBatch);
+  }
+
+  private void sortRowBatch(RowBatch rowBatch){
+    /*
+       * following part of code sort the batch data by time,
+       * so we can insert continuous data in value list to get a better performance
+       */
+    // sort to get index, and use index to sort value list
```

Comment:
I would recommend that you check if the batch is already sorted before doing the actual sort.

Reply:
fixed~